src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java

```diff
@@ -0,0 +1,104 @@
+     * - moves files to container for rejected files
+     * - removes files from the original container
+     * - marks envelopes in the DB as deleted
+     */
+    public void handle() {
+        List<Envelope> rejectedEnvelopes = envelopeRepository.find(Status.REJECTED, false);
+
+        logger.info("Found {} rejected envelopes", rejectedEnvelopes.size());
+
+        rejectedEnvelopes
+            .stream()
+            .collect(groupingBy(e -> e.container))
+            .keySet()
+            .forEach(container -> {
+
+                logger.info("Moving rejected files from container {}", container);
+
+                var sourceContainer = storageClient.getBlobContainerClient(container);
+                var targetContainer = storageClient.getBlobContainerClient(container + REJECTED_CONTAINER_SUFFIX);
+
+                rejectedEnvelopes
+                    .stream()
+                    .collect(groupingBy(env -> env.container))
+                    .get(container)
+                    .forEach(envelope -> {
+                        try {
+                            BlobClient sourceBlob = sourceContainer.getBlobClient(envelope.fileName);
+                            BlobClient targetBlob = targetContainer.getBlobClient(envelope.fileName);
+
+                            if (!sourceBlob.exists()) {
+                                logger.error(
+                                    "File already deleted. File name: {}. Container: {}",
+                                    envelope.fileName,
+                                    container
+                                );
+                                envelopeRepository.markAsDeleted(envelope.id);
+                            } else {
+                                byte[] blobContent = download(sourceBlob);
+                                upload(targetBlob, blobContent);
+                                sourceBlob.delete();
+                                envelopeRepository.markAsDeleted(envelope.id);
+                            }
+
+                        } catch (Exception exc) {
+                            logger.error(
+                                "Error handling rejected file. File name: {}. Container: {}",
+                                envelope.fileName,
+                                container,
+                                exc
+                            );
+                        }
+                    });
+            });
+    }
+
+    private byte[] download(BlobClient blobClient) throws IOException {
+        try (var outputStream = new ByteArrayOutputStream()) {
+            blobClient.download(outputStream);
+            return outputStream.toByteArray();
+        }
+    }
+
+    private void upload(BlobClient blobClient, byte[] blobContent) {
+        blobClient
+            .getBlockBlobClient()
+            .upload(
+                new ByteArrayInputStream(blobContent),
+                blobContent.length
+            );
+    }
```

Comment:
We're downloading in order to upload, we could avoid having byte arrays and use steams instead. We can't do this in the main job, as we need to verify the signature and extract the content, but it's possible here.